ts/lib/carina.ts

```diff
@@ -53,14 +59,15 @@ export class Carina {
      * @returns {Promise.<>} Resolves once subscribed. Any errors will reject.
      */
     public subscribe<T>(slug: string, cb: (data: T) => void): Promise<any> {
-        this.socket.on('event:live', data => {
+        this.socket.on('event:live', (data: { channel: string, payload: any }) => {
             if (data.channel === slug) {
                 cb(data.payload);
             }
         });
 
         return this
         .waitFor(`subscription:${slug}`, () => {
+            this.subscriptions.push(slug);
```

Comment:
So you want? 
```javascript
    return this
        .waitFor(`subscription:${slug}`, () => {
            return this.socket.execute('livesubscribe', { events: [slug] })
            .then(res => {
                this.subscriptions.push(slug);
                return res;
            });
        })
        .catch(err => {
            this.stopWaiting(`subscription:${slug}`);
            throw err;
        });
```
Rather than the `then` after?